components/ModEditFacilityOrProfessionalTopbar.vue

```diff
@@ -178,17 +197,55 @@ const facilityHasUnsavedChanges = () => {
         !== JSON.stringify(facilitySections.healthcareProfessionalIds)
         || facilitySections.healthProfessionalsRelations.length
 
-    return areThereUnsavedChanges
+        return areThereUnsavedChanges
+    }
+    if (moderationScreenStore.activeScreen === ModerationScreen.EditHealthcareProfessional) {
```

Comment:
We perform these active screen checks in a lot of places in the repo. At some point (and not necessarily this PR) we should write functions to make them a bit cleaner. IE `if (activeScreenIsEditHealthcareProfessional())`.